src/viur/datastore/transport.pyx

```diff
@@ -842,9 +824,10 @@ def RunInTransaction(callback: callable, *args, **kwargs) -> Any:
 						return res
 				finally:  # Ensure, currentTransaction is always set back to none
 					currentTransaction.set(None)
-		except Collision:  # Got a collision; retry the entire transaction
-			logging.warning("Transaction collision, retrying...")
-	raise Collision() # If we made it here, all tries are exhausted
+		except CollisionError:  # Got a collision; retry the entire transaction
+			sleep(exponential_backoff ** 2)
```

Comment:
This is not an _exponential_ backoff, this is a _square_ backoff.

```suggestion
			sleep(2 ** exponential_backoff)
```

Furthermore we would we would wait 0 seconds after the first fail with square and 1 second with exponential backoff. Not sure if the immediate retry is intended.